extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ApplyRouteTargetPortConfigurator.java

```diff
@@ -0,0 +1,31 @@
+package io.quarkus.kubernetes.deployment;
+
+import io.dekorate.kubernetes.decorator.NamedResourceDecorator;
+import io.fabric8.kubernetes.api.model.IntOrString;
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.openshift.api.model.RoutePort;
+import io.fabric8.openshift.api.model.RouteSpecFluent;
+
+public class ApplyRouteTargetPortConfigurator extends NamedResourceDecorator<RouteSpecFluent<?>> {
+
+    private IntOrString targetPort;
+
+    public ApplyRouteTargetPortConfigurator(Integer port) {
+        super(ANY);
+        targetPort = new IntOrString(port);
+    }
+
+    public ApplyRouteTargetPortConfigurator(String port) {
+        super(ANY);
+        try {
```

Comment:
I don't like coding by exception but struggled to think of an alternative here to ensuring that numerical values are treated properly. Without this, if the value in the property is set to a number it is written to the yml file quoted as a string regardless. I believe this will have no functional impact, but would still represent an undesirable change to how it currently works. Any thoughts/suggestions on how to approach this?

Reply:
I don't see any no problem in having it written as a String all the time